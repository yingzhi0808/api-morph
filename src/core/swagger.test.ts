import { describe, expect, it } from "vitest";
import {
  type GenerateSwaggerUIHTMLOptions,
  generateSwaggerUIHTML,
  getSwaggerUIAssetDir,
} from "./swagger";

describe("swagger", () => {
  describe("getSwaggerUIAssetPaths", () => {
    it("应该正确返回Swagger UI静态资源目录", () => {
      const assetDir = getSwaggerUIAssetDir();
      expect(assetDir).toContain("swagger-ui-dist");
    });
  });

  describe("generateSwaggerUIHTML", () => {
    it("应该使用默认选项生成HTML", () => {
      const html = generateSwaggerUIHTML();

      expect(html).toMatchSnapshot();
    });

    it("应该正确应用自定义URL", () => {
      const options: GenerateSwaggerUIHTMLOptions = {
        url: "/api/docs/openapi.json",
      };
      const html = generateSwaggerUIHTML(options);

      expect(html).toMatchSnapshot();
    });

    it("应该正确应用自定义标题", () => {
      const options: GenerateSwaggerUIHTMLOptions = {
        title: "My Custom API Documentation",
      };
      const html = generateSwaggerUIHTML(options);

      expect(html).toMatchSnapshot();
    });

    it("应该正确应用自定义CSS", () => {
      const customCss = `
        body { background-color: #f0f0f0; }
        .swagger-ui { font-family: Arial; }
      `;
      const options: GenerateSwaggerUIHTMLOptions = {
        customCss,
      };
      const html = generateSwaggerUIHTML(options);

      expect(html).toMatchSnapshot();
    });

    it("应该正确应用自定义JavaScript", () => {
      const customJs = `
        console.log("Custom JavaScript loaded");
        window.customFunction = function() { return "test"; };
      `;
      const options: GenerateSwaggerUIHTMLOptions = {
        customJs,
      };
      const html = generateSwaggerUIHTML(options);

      expect(html).toMatchSnapshot();
    });

    it("应该正确设置persistAuthorization为true", () => {
      const options: GenerateSwaggerUIHTMLOptions = {
        persistAuthorization: true,
      };
      const html = generateSwaggerUIHTML(options);

      expect(html).toMatchSnapshot();
    });
  });
});
