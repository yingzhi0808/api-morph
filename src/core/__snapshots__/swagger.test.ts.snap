// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`swagger > generateSwaggerUIHTML > 应该使用默认选项生成HTML 1`] = `
"<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>Swagger UI</title>
  <link rel="icon" type="image/png" href="./favicon-32x32.png" sizes="32x32">
  <link rel="icon" type="image/png" href="./favicon-16x16.png" sizes="16x16">
  <link rel="stylesheet" href="./swagger-ui.css">
  <link rel="stylesheet" href="./index.css">
  
</head>
<body>
  <div id="swagger-ui"></div>
  <script src="./swagger-ui-bundle.js"></script>
  <script src="./swagger-ui-standalone-preset.js"></script>
  <script>
    window.onload = () => {
      const ui = SwaggerUIBundle({
        url: "/openapi.json",
        dom_id: "#swagger-ui",
        deepLinking: true,
        presets: [
          SwaggerUIBundle.presets.apis,
          SwaggerUIStandalonePreset
        ],
        plugins: [
          SwaggerUIBundle.plugins.DownloadUrl
        ],
        layout: "StandaloneLayout",
        persistAuthorization: false
      });

      window.ui = ui;
    };
    
  </script>
</body>
</html>"
`;

exports[`swagger > generateSwaggerUIHTML > 应该正确应用自定义CSS 1`] = `
"<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>Swagger UI</title>
  <link rel="icon" type="image/png" href="./favicon-32x32.png" sizes="32x32">
  <link rel="icon" type="image/png" href="./favicon-16x16.png" sizes="16x16">
  <link rel="stylesheet" href="./swagger-ui.css">
  <link rel="stylesheet" href="./index.css">
  <style>

        body { background-color: #f0f0f0; }
        .swagger-ui { font-family: Arial; }
      
</style>
</head>
<body>
  <div id="swagger-ui"></div>
  <script src="./swagger-ui-bundle.js"></script>
  <script src="./swagger-ui-standalone-preset.js"></script>
  <script>
    window.onload = () => {
      const ui = SwaggerUIBundle({
        url: "/openapi.json",
        dom_id: "#swagger-ui",
        deepLinking: true,
        presets: [
          SwaggerUIBundle.presets.apis,
          SwaggerUIStandalonePreset
        ],
        plugins: [
          SwaggerUIBundle.plugins.DownloadUrl
        ],
        layout: "StandaloneLayout",
        persistAuthorization: false
      });

      window.ui = ui;
    };
    
  </script>
</body>
</html>"
`;

exports[`swagger > generateSwaggerUIHTML > 应该正确应用自定义JavaScript 1`] = `
"<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>Swagger UI</title>
  <link rel="icon" type="image/png" href="./favicon-32x32.png" sizes="32x32">
  <link rel="icon" type="image/png" href="./favicon-16x16.png" sizes="16x16">
  <link rel="stylesheet" href="./swagger-ui.css">
  <link rel="stylesheet" href="./index.css">
  
</head>
<body>
  <div id="swagger-ui"></div>
  <script src="./swagger-ui-bundle.js"></script>
  <script src="./swagger-ui-standalone-preset.js"></script>
  <script>
    window.onload = () => {
      const ui = SwaggerUIBundle({
        url: "/openapi.json",
        dom_id: "#swagger-ui",
        deepLinking: true,
        presets: [
          SwaggerUIBundle.presets.apis,
          SwaggerUIStandalonePreset
        ],
        plugins: [
          SwaggerUIBundle.plugins.DownloadUrl
        ],
        layout: "StandaloneLayout",
        persistAuthorization: false
      });

      window.ui = ui;
    };
    
        console.log("Custom JavaScript loaded");
        window.customFunction = function() { return "test"; };
      
  </script>
</body>
</html>"
`;

exports[`swagger > generateSwaggerUIHTML > 应该正确应用自定义URL 1`] = `
"<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>Swagger UI</title>
  <link rel="icon" type="image/png" href="./favicon-32x32.png" sizes="32x32">
  <link rel="icon" type="image/png" href="./favicon-16x16.png" sizes="16x16">
  <link rel="stylesheet" href="./swagger-ui.css">
  <link rel="stylesheet" href="./index.css">
  
</head>
<body>
  <div id="swagger-ui"></div>
  <script src="./swagger-ui-bundle.js"></script>
  <script src="./swagger-ui-standalone-preset.js"></script>
  <script>
    window.onload = () => {
      const ui = SwaggerUIBundle({
        url: "/api/docs/openapi.json",
        dom_id: "#swagger-ui",
        deepLinking: true,
        presets: [
          SwaggerUIBundle.presets.apis,
          SwaggerUIStandalonePreset
        ],
        plugins: [
          SwaggerUIBundle.plugins.DownloadUrl
        ],
        layout: "StandaloneLayout",
        persistAuthorization: false
      });

      window.ui = ui;
    };
    
  </script>
</body>
</html>"
`;

exports[`swagger > generateSwaggerUIHTML > 应该正确应用自定义标题 1`] = `
"<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>My Custom API Documentation</title>
  <link rel="icon" type="image/png" href="./favicon-32x32.png" sizes="32x32">
  <link rel="icon" type="image/png" href="./favicon-16x16.png" sizes="16x16">
  <link rel="stylesheet" href="./swagger-ui.css">
  <link rel="stylesheet" href="./index.css">
  
</head>
<body>
  <div id="swagger-ui"></div>
  <script src="./swagger-ui-bundle.js"></script>
  <script src="./swagger-ui-standalone-preset.js"></script>
  <script>
    window.onload = () => {
      const ui = SwaggerUIBundle({
        url: "/openapi.json",
        dom_id: "#swagger-ui",
        deepLinking: true,
        presets: [
          SwaggerUIBundle.presets.apis,
          SwaggerUIStandalonePreset
        ],
        plugins: [
          SwaggerUIBundle.plugins.DownloadUrl
        ],
        layout: "StandaloneLayout",
        persistAuthorization: false
      });

      window.ui = ui;
    };
    
  </script>
</body>
</html>"
`;

exports[`swagger > generateSwaggerUIHTML > 应该正确设置persistAuthorization为true 1`] = `
"<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>Swagger UI</title>
  <link rel="icon" type="image/png" href="./favicon-32x32.png" sizes="32x32">
  <link rel="icon" type="image/png" href="./favicon-16x16.png" sizes="16x16">
  <link rel="stylesheet" href="./swagger-ui.css">
  <link rel="stylesheet" href="./index.css">
  
</head>
<body>
  <div id="swagger-ui"></div>
  <script src="./swagger-ui-bundle.js"></script>
  <script src="./swagger-ui-standalone-preset.js"></script>
  <script>
    window.onload = () => {
      const ui = SwaggerUIBundle({
        url: "/openapi.json",
        dom_id: "#swagger-ui",
        deepLinking: true,
        presets: [
          SwaggerUIBundle.presets.apis,
          SwaggerUIStandalonePreset
        ],
        plugins: [
          SwaggerUIBundle.plugins.DownloadUrl
        ],
        layout: "StandaloneLayout",
        persistAuthorization: true
      });

      window.ui = ui;
    };
    
  </script>
</body>
</html>"
`;
